tests/dragonfly/replication_test.py

```diff
@@ -2661,3 +2661,23 @@ async def check_master_status():
     assert await seeder.compare(capture, port=master.port)
 
     await disconnect_clients(c_master, c_replica)
+
+
+@pytest.mark.asyncio
+async def test_bug_3528(df_factory):
+    master = df_factory.create()
+    replica = df_factory.create()
+    df_factory.start_all([master, replica])
+
+    c_replica = replica.client()
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    c_master = master.client()
+    await c_master.execute_command(f"HSET foo bar val")
```

Comment:
I would add this as a subtest case of def test_rewrites

Reply:
you will check there that del foo was sent

Reply:
hmm I did not know that `bitops` acts as a blind update. I tested this with Valkey, so if the operation returns `WRONG_TYPE` we should still `delete the key`. I will add this to the logic + comment + move this to rewrite

Reply:
So the fix above is also incorrect `if (find_res == OpStatus::OK)`!